MaterialLib/Fluid/Density/createFluidDensityModel.cpp

```diff
@@ -0,0 +1,106 @@
+/*!
+   \file  createFluidDensityModel.cpp
+   \brief create an instance of a fluid density class.
+
+   \copyright
+    Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+               Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+*/
+
+#include <array>
+
+#include "createFluidDensityModel.h"
+
+#include "BaseLib/Error.h"
+
+#include "MaterialLib/Fluid/ConstantFluidProperty.h"
+#include "IdealGasLaw.h"
+#include "LinearTemperatureDependentDensity.h"
+#include "LiquidDensity.h"
+
+namespace MaterialLib
+{
+namespace Fluid
+{
+/*
+    config  ConfigTree object which contains the input data
+                   including  <type>fluid</type> and it has
+                   a tag of <density>
+*/
+std::unique_ptr<FluidProperty> createLiquidDensity(BaseLib::ConfigTree const& config)
+{
+    std::array<double, 5> parameters =
+    {
+        //! \ogs_file_param{material__fluid__density__liquid_density__beta}
+        config.getConfigParameter<double>("beta"),
+        //! \ogs_file_param{material__fluid__density__liquid_density__rho0}
+        config.getConfigParameter<double>("rho0"),
+        //! \ogs_file_param{material__fluid__density__liquid_density__temperature0}
+        config.getConfigParameter<double>("temperature0"),
+        //! \ogs_file_param{material__fluid__density__liquid_density__p0}
+        config.getConfigParameter<double>("p0"),
+        //! \ogs_file_param{material__fluid__density__liquid_density__bulk_modulus}
+        config.getConfigParameter<double>("bulk_modulus")
+    };
+    return std::unique_ptr<FluidProperty>(new LiquidDensity(parameters));
+}
+
+/*
+    config  ConfigTree object which contains the input data
+                   including  <type>fluid</type> and it has
+                   a tag of <density>
+*/
+std::unique_ptr<FluidProperty> createLinearTemperatureDependentDensity
```

Comment:
Adding a `static` specifier will make explicit, that the function is used only in this translation unit. This is an actual clang warning, about missing declaration.